util/file_reader_writer.h

```diff
@@ -109,6 +112,8 @@ class RandomAccessFileReader {
 
   RandomAccessFile* file() { return file_.get(); }
 
+  std::string file_name() const { return file_name_; }
```

Comment:
I thought of that but decided that it makes more sense to put this responsibility on a non-abstract class that's internal to rocksdb codebase rather than on an abstract class exposed to the users. Also the existing custom RandomAccessFile implementations would need to be updated (especially the ones outside rocksdb codebase). (I guess file_name() would have a default implementation that returns an empty string or something, but then again the users with custom RandomAccessFile's would need to update them to get the file names in Status messages.)